pgmpy/factors/CPD.py

```diff
@@ -107,27 +107,25 @@ def __init__(self, variable, variable_card, values,
 
         cardinality = [variable_card]
         if evidence_card is not None:
-            if not isinstance(evidence_card, (list, set, tuple)):
+            if not isinstance(evidence_card, (list, tuple)):
                 if isinstance(evidence_card, np.ndarray):
                     evidence_card = evidence_card.tolist()
                 elif isinstance(evidence_card, (int, float)):
                     evidence_card = [evidence_card]
                 else:
-                    raise TypeError("Evidence cardinality must be list, set, "
+                    raise TypeError("Evidence cardinality must be list, "
                                     "tuple, numpy ndarray or simply a number.")
-            self.evidence_card = evidence_card
             cardinality.extend(evidence_card[::-1])
```

Comment:
@anaviltripathi What I understand now (not very sure) is that the earlier representation was considering the fastest changing parent first. So, if the evidence was `['B', 'C']` then the table should have been something like this:
```
+-----+-----+-----+-----+-----+-----+-----+
| C   | C_0 | C_1 | C_0 | C_1 | C_0 | C_1 |
+-----+-----+-----+-----+-----+-----+-----+
| B   | B_0 | B_0 | B_1 | B_1 | B_2 | B_2 |
+-----+-----+-----+-----+-----+-----+-----+
+-----+-----+-----+-----+-----+-----+-----+
```
But the `__str__` method wasn't considering this.

Anyways let's fix this all now. I think the earlier representation of evidence was not that intuitive. And intuitive way for defining evidence for the above table should be `['C', 'B']`. What do you think ?